tensorboard/webapp/widgets/line_chart_v2/lib/formatter.ts

```diff
@@ -0,0 +1,122 @@
+/* Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+import {format, scaleTime} from '../../../third_party/d3';
+
+export interface Formatter {
+  formatTick(x: number): string;
+  formatShort(x: number): string;
+  formatLong(x: number): string;
```

Comment:
It looks like the idea here is that `formatShort` is allowed to be lossy
and imprecise (`100001` → `1e5`) whereas `formatLong` is supposed to
give an ~exact value at the cost of a potentially longer string. Is that
right? If so, could you briefly document it?